statsmodels/kernel_methods/tests/test_fastbin.py

```diff
@@ -0,0 +1,247 @@
+from __future__ import division, absolute_import, print_function
+
+from .. import _fast_linbin as linbin
+
+from nose.plugins.attrib import attr
+from nose.tools import raises
+from ...tools.testing import (assert_equal, assert_allclose)
+from scipy import stats
+from itertools import product
+import numpy as np
+
+@attr("kernel_methods")
+class TestContinuousBinning1D(object):
+    @classmethod
+    def setUpClass(cls):
+        dst = stats.norm(0, 1)
+        cls.data = dst.rvs(2000)
+        cls.weights = stats.uniform(1, 5).rvs(2000)
+        cls.bin_types = 'CRB'
+        cls.bounds = [-3, 3]
+        cls.sizes = (64, 128, 159)
+
+    def validity(self, fct, M, bin_type, weighted):
+        data = self.data
+        bounds = self.bounds
+        weights = 1.
+        size = data.shape[0]
+        if bin_type == 'B':
+            sel = (data >= bounds[0]) & (data < bounds[1])
+            if weighted:
+                weights = self.weights
+                size = weights[sel].sum()
+            else:
+                size = sel.sum()
+        elif weighted:
+            weights = self.weights
+            size = weights.sum()
+        mesh, bins = fct(data, bounds, M, weights, bin_type)
+        assert_equal(mesh.shape, (M,))
+        assert mesh.grid[0][0] >= bounds[0]
+        assert mesh.grid[0][-1] <= bounds[1]
+        assert_allclose(bins.sum(), size, rtol=1e-8)
+
+    def test_validity(self):
+        for fct, s, t in product([linbin.fast_bin, linbin.fast_linbin],
+                                 self.sizes, self.bin_types):
+            yield self.validity, fct, s, t, True
+            yield self.validity, fct, s, t, False
+
+    @raises(ValueError)
+    def bad_data(self, fct):
+        bounds = self.bounds
+        fct([[[1]]], bounds, 256, 1., 'B')
+
+    @raises(ValueError)
+    def bad_bin_type1(self, fct):
+        bounds = self.bounds
+        fct(self.data, bounds, 256, 1., 'X')
+
+    @raises(ValueError)
+    def bad_bin_type2(self, fct):
+        bounds = self.bounds
+        fct(self.data, bounds, 256, 1., 'BB')
+
+    @raises(ValueError)
+    def bad_bin_type3(self, fct):
+        bounds = self.bounds
+        fct(self.data, bounds, 256, 1., '')
+
+    @raises(ValueError)
+    def bad_weights(self, fct):
+        bounds = self.bounds
+        fct(self.data, bounds, 256, [1, 2, 3], 'B')
+
+    @raises(ValueError)
+    def bad_bounds1(self, fct):
+        fct(self.data, [[1, 2], [3, 4]], 256, 1., 'B')
+
+    @raises(ValueError)
+    def bad_bounds2(self, fct):
+        fct(self.data, [1, 2, 3], 256, 1., 'B')
+
+    @raises(ValueError)
+    def bad_bounds3(self, fct):
+        fct(self.data, 'bad', 256, 1., 'B')
+
+    @raises(ValueError)
+    def bad_out1(self, fct):
+        out = np.empty((1,), dtype=float)
+        fct(self.data, self.bounds, 256, 1., 'B', out)
+
+    @raises(ValueError)
+    def bad_out2(self, fct):
```

Comment:
add third argument here, or just *args

to take name.  This will allow verbose printing to show information to identify test failures